src/fprime/fpp/impl.py

```diff
@@ -0,0 +1,163 @@
+""" fprime.fpp.impl: Command line targets for `fprime-util impl`
+
+Processing and CLI entry points for `fprime-util impl` command line tool.
+
+@author thomas-bc
+"""
+
+
+import argparse
+import os
+import tempfile
+from pathlib import Path
+
+from typing import TYPE_CHECKING, Callable, Dict, List, Tuple
+
+if TYPE_CHECKING:
+    from fprime.fbuild.builder import Build
+
+from fprime.fpp.common import FppUtility
+from fprime.util.code_formatter import ClangFormatter
+
+
+def fpp_generate_implementation(
+    build: "Build",
+    output_dir: Path,
+    context: Path,
+    apply_formatting: bool,
+    ut: bool,
+    auto_test_helpers: bool = False,
+) -> int:
+    """
+    Generate implementation files from FPP templates.
+
+    Args:
+        build: Build object
+        output_dir: The directory where the generated files will be written
+        context: The path to the F´ module to generate files for
+        apply_formatting: Whether to format the generated files using clang-format
+        ut: Generates UT files if set to True
+    """
+
+    prefixes = [
+        build.get_settings("framework_path", ""),
+        *build.get_settings("library_locations", []),
+        build.cmake_root,
+        build.build_dir / "F-Prime",
+        build.build_dir,
+    ]
+
+    # Holds the list of generated files to be passed to clang-format
+    gen_files = tempfile.NamedTemporaryFile(prefix="fprime-impl-")
+
+    output_path = Path(output_dir)
+    if ut:
+        output_path = output_path / "test/ut"
```

Comment:
This will cause clobbering. @bocchino recommends we generate in the component (not UT) directory until FPP adds `.template` to the UT impl outputs.  Then we can change.